crates/burn-jit/src/codegen/compilation.rs

```diff
@@ -193,7 +193,10 @@ impl CompilationSettings {
                     if chosen.is_some() {
                         break;
                     }
-                    if desc.shape == desc_input.shape && input.item() == output.item() {
+                    if desc.shape == desc_input.shape
+                        && input.item() == output.item()
+                        && desc.id == desc_input.id
+                    {
```

Comment:
I don't think that's right, it's not a requirement! But I think we should verify before chosing an input to replace the output if the output is also an input, in which case we have to use it.